datasketch/weighted_minhash.py

```diff
@@ -1,6 +1,12 @@
+from typing import Union, List
+
 import collections
+import collections.abc
 import copy
+
 import numpy as np
+import scipy as sp
```

Comment:
This project currently has only one hard dependency that is Numpy. Scipy is still optional. Is it possible to make it optional for WeightMinHash? So the method "minhash_many" is only available when Scipy is present. See here for example: https://github.com/ekzhu/datasketch/blob/master/datasketch/lsh.py#L18

Reply:
It would be tricky, but with some dtype and indexing tricks it might be possible to re-implement the code without scipy.sparse.  I'll give it a shot.

Reply:
I was thinking about raising error when invoking the method rather than importing the module. Anyway, I think at this stage Scipy is often available already and we are using well-established Scipy method here we can include Scipy as a dependency.